core-service/src/test/java/com/crmfoodestablishment/coreservice/controller/MenuControllerTest.java

```diff
@@ -0,0 +1,173 @@
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.any;
+import static org.mockito.Mockito.when;
+
+@WebMvcTest(MenuController.class)
+@ExtendWith(MockitoExtension.class)
+class MenuControllerTest {
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @MockBean
+    private MenuServiceImpl menuServiceImpl;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    private MenuDto createMenuDto() {
+        MenuDto menuDto = new MenuDto();
+        menuDto.setUuid(UUID.randomUUID());
+        menuDto.setName("summer menu");
+        menuDto.setComment("new summer menu");
+        menuDto.setSeason(Season.SUMMER);
+        return menuDto;
+    }
+
+    @Test
+    void shouldCreateMenu() throws Exception {
+        MenuDto menuDto = createMenuDto();
+
+        when(menuServiceImpl.addMenu(any(MenuDto.class))).thenReturn(menuDto.getUuid());
+
+        mockMvc.perform(MockMvcRequestBuilders.post("/api/menu")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(objectMapper.writeValueAsString(menuDto)))
+                .andExpect(MockMvcResultMatchers.status().isCreated())
+                .andExpect(content().contentType(MediaType.APPLICATION_JSON))
+                .andExpect(jsonPath("$").value(menuDto.getUuid().toString()));
+    }
+
+    @Test
+    void shouldThrow400BadRequestWhenRequestDtosNotValid() throws Exception {
+        MenuDto menuDto = new MenuDto();
+        menuDto.setName(null);
+        menuDto.setComment("new summer menu");
+        menuDto.setSeason(Season.SUMMER);
+
+        mockMvc.perform(MockMvcRequestBuilders.post("/api/menu")
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(objectMapper.writeValueAsString(menuDto)))
+                .andExpect(status().isBadRequest())
+                .andExpect(content().contentType(MediaType.APPLICATION_JSON))
+                .andExpect(jsonPath("$.currentTime").isNotEmpty())
+                .andExpect(jsonPath("$.status").value(400))
+                .andExpect(jsonPath("$.error").value("Validation failed"))
+                .andExpect(jsonPath("$.errors['name']").value("Field name cannot be blank"));
+    }
+
+    @Test
+    void shouldGetValidInputThenMapsToBusinessModel() throws Exception {
+        MenuDto menuDto = createMenuDto();
+
+        mockMvc.perform(MockMvcRequestBuilders.post("/api/menu")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(objectMapper.writeValueAsString(menuDto)))
+                .andExpect(MockMvcResultMatchers.status().isCreated());
+        ArgumentCaptor<MenuDto> argumentCaptor = ArgumentCaptor.forClass(MenuDto.class);
+
+        verify(menuServiceImpl, times(1)).addMenu(argumentCaptor.capture());
+        MenuDto menuCaptorValue = argumentCaptor.getValue();
```

Comment:
pls rename this 'menuCaptorValue', it is a bit unclear. It is better to use 'actualMenu' or smth like this